dataplicity/m2m/wsclient.py

```diff
@@ -407,7 +413,11 @@ def send_bytes(self, packet_bytes):
             if self.app.client_terminated:
                 return False
             else:
-                self.app.send(packet_bytes, binary=True)
+                try:
```

Comment:
All packets go through here. If there is a socket error, we can only attempt to recover by terminating and restarting.